interop-testing/src/test/java/io/grpc/testing/integration/NettyFlowControlTest.java

```diff
@@ -0,0 +1,374 @@
+  private static final TrafficControlProxy proxy =
+      new TrafficControlProxy(LOW_BAND, LOW_LAT, TimeUnit.MILLISECONDS);
+
+  // HandlerAccessor accessor = new HandlerAccessor();
+
+  /*
+   * Keeping this main class around for now because it's convenient to be able to run this from the
+   * command line.
+   */
+  public static void main(String[] args) throws UnknownHostException {
+    NettyFlowControlTest ft = new NettyFlowControlTest();
+    setUp();
+    ft.lowBandLowLatency();
+    ft.lowBandHighLatency();
+    ft.highBandLowLatency();
+    ft.highBandHighLatency();
+    ft.verySmallWindow();
+    shutDownTests();
+    ft.printResults();
+  }
+
+  public void printResults() {
+    System.out.println("Passed: " + testsPassed + " Failed: " + testsFailed);
+  }
+
+  @BeforeClass
+  public static void setUp() throws UnknownHostException {
+    HandlerSettings.autoWindowOn(true);
+    addr = InetAddress.getByName(loopback);
+    logger = Logger.getLogger("io.grpc.netty.NettyClientHandler");
+    logger.setLevel(Level.FINEST);
+    logHandler = new TestLogHandler();
+    logger.addHandler(logHandler);
+    startProxy();
+  }
+
+  @AfterClass
+  public static void shutDownTests() {
+    try {
+      proxy.shutDown();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    executor.shutdown();
+    server.shutdown();
+  }
+
+
+  @Test
+  public void doTests() {
+    lowBandLowLatency();
+    lowBandHighLatency();
+    highBandLowLatency();
+    highBandHighLatency();
+    verySmallWindow();
+    assertEquals(0, testsFailed);
+  }
+
+  public void endTest() {
+    server.shutdown();
+  }
+
+  // It is necessary to restart the server every time in order to bring the window back down.
+  private void lowBandLowLatency() {
+    resetConnection(REGULAR_WINDOW, REGULAR_WINDOW);
+    doTest(LOW_BAND, LOW_LAT);
+    endTest();
+  }
+
+  private void lowBandHighLatency() {
+    resetProxy(LOW_BAND, HIGH_LAT, TimeUnit.MILLISECONDS);
+    resetConnection(REGULAR_WINDOW, REGULAR_WINDOW);
+    doTest(LOW_BAND, HIGH_LAT);
+    endTest();
+  }
+
+  private void highBandLowLatency() {
+    resetProxy(HIGH_BAND, LOW_LAT, TimeUnit.MILLISECONDS);
+    resetConnection(REGULAR_WINDOW, REGULAR_WINDOW);
+    doTest(HIGH_BAND, LOW_LAT);
+    endTest();
+  }
+
+  private void highBandHighLatency() {
+    resetProxy(HIGH_BAND, HIGH_LAT, TimeUnit.MILLISECONDS);
+    resetConnection(REGULAR_WINDOW, REGULAR_WINDOW);
+    doTest(HIGH_BAND, HIGH_LAT);
+    endTest();
+  }
+
+  private void verySmallWindow() {
+    resetProxy(MED_BAND, MED_LAT, TimeUnit.MILLISECONDS);
+    resetConnection(REGULAR_WINDOW, TINY_WINDOW);
+    doTest(MED_BAND, MED_LAT);
+    endTest();
+  }
+
+  /**
+   * Main testing method. Streams 4 MB of data from a server and records the final window and
+   * average bandwidth usage
+   *
+   * @param bandwidth
+   * @param latency
+   */
+  private void doTest(int bandwidth, int latency) {
```

Comment:
It's okay in tests to just throw Exception (because the test runner will just note the failure, which is what we want).

That would alleviate the need to have try catches below.